packages/shared/src/api/lanyardApi.ts

```diff
@@ -82,32 +92,91 @@ export async function discoverContacts(phoneNums: string[]) {
     try {
       await pokeNoun({ app: 'lanyard', mark: 'lanyard-query-1', noun });
     } catch (e) {
-      console.error('contact discovery poke error', e);
-    }
-    const queryResponse = await queryResponseSub;
-    console.log(`bulk result`, queryResponse);
-
-    if (queryResponse) {
-      // return matches
-      const nextSalt = queryResponse.result?.['next-salt'];
-      const matches = queryResponse.result?.result
-        ? Object.entries(queryResponse.result.result).filter(([_key, value]) =>
-            Boolean(value)
-          )
-        : [];
-
-      return matches;
+      logger.trackError('contact discovery poke error', e);
+      logger.trackEvent(AnalyticsEvent.DebugContactMatching, {
```

Comment:
Instead of double logging here, can we add a `ErrorContactMatching` event type and tack on a `context` property to signal where it failed? 

See `DebugSystemContacts` and `ErrorSystemContacts` as an example.